lib/schema.js

```diff
@@ -375,14 +391,15 @@ function RecordSchema(name, namespace, fields) {
 
 util.inherits(RecordSchema, Schema);
 
-function MapSchema(type) {
+function MapSchema(type, defaultValue) {
     this.type = 'map';
     this.values = type;
+    this.default = defaultValue;
```

Comment:
Do all schema classes have a default value?  What about storing defaultValue in the Schema class and invoking Schema.call?  I realize this doesn't save much here, but seems to be the right thing looking forward.